src/processes/helper/sendTopicAutoMessage.js

```diff
@@ -4,21 +4,29 @@ const {sample} = require('lodash');
 const {followTopicQueue} = require('../../config/redis');
 const {CommunityMessageFormat, LogError} = require('../../databases/models');
 
+const setRequiredTime = (delay) => {
+  let requiredTime = momentTz().tz('America/Los_Angeles');
+  if (delay === 0) {
+    const randomTime = Math.floor(Math.random() * (40 - 10 + 1)) + 10;
+    requiredTime = momentTz().tz('America/Los_Angeles').add(randomTime, 'minutes');
+  } else {
+    const randomTime = sample([6, 7, 8]);
+    const additionalDays = delay;
+    requiredTime = momentTz()
+      .tz('America/Los_Angeles')
+      .set({hour: randomTime})
+      .add(additionalDays, 'days');
+  }
+  return requiredTime;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `setRequiredTime` function seems to be working as expected. However, it's not clear what the `sample` function does on line 13. If it's a custom function, make sure it's imported or defined in this file. If it's meant to pick a random element from the array, you might want to use a more standard approach like picking a random index for the array.

```diff
-    const randomTime = sample([6, 7, 8]);
+    const randomTime = [6, 7, 8][Math.floor(Math.random() * 3)];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->